### PR TITLE
fix: facebook channels

### PR DESCRIPTION
**Pull request by axelazaid**
* updated the facebook ads channel to account for facebook ads coming through amazon

### CODE REVIEW

File: models/marts/advertising/fct_ad_performance.sql [hunk 1]
```diff
@@ -47,14 +47,19 @@ marketing_channels as (
...
-
+        case 
             when lower(campaign) ilike '%display%' then 'display'
             when lower(campaign) ilike '%shopping%'
-                or lower(campaign) ilike '%cpc%' then 'paid search' 
-            when utm_medium is null then 'direct'
-            when lower(utm_medium) in ('retargeting', 'social', 'influencers') then utm_medium
-            when lower(utm_medium) in ('paid', 'paid_social') then 'facebook ads'
+                or lower(campaign) ilike '%cpc%' then 'paid search'
+            when utm_medium is null and utm_medium not in ('fb_retargeted', 'ig_retargeted')
```
Comment (eogilvy12) [inline models/marts/advertising/fct_ad_performance.sql]: the logic here doesn't make sense to me. something cannot be null and be `in` something. is this supposed to be an OR statement?

File: models/marts/advertising/fct_ad_performance.sql [hunk 2]
```diff
@@ -47,14 +47,19 @@ marketing_channels as (
...
+                or lower(campaign) ilike '%cpc%' then 'paid search'
+            when utm_medium is null and utm_medium not in ('fb_retargeted', 'ig_retargeted') 
+                then 'direct'
+            when lower(utm_medium) in ('retargeting', 'social', 'influencers') 
+                then utm_medium
+            when lower(utm_campaign) in ('fb_retargeted', 'ig_retargeted') 
+                then 'retargeting'
+            when lower(utm_medium) in ('paid', 'paid_social') 
+                and lower(utm_campaign) not ilike '%amazon%' then 'facebook ads'
+            when lower(utm_campaign) ilike '%amazon%' then 'facebook amazon'
```
Comment (eogilvy12) [inline models/marts/advertising/fct_ad_performance.sql:58]: if you put the amazon logic first you will not have to not ilike amazon in the `facebook ads` logic

**Review by eogilvy12 — CHANGES_REQUESTED** [review]

File: models/marts/advertising/fct_ad_performance.sql (hunk 1 above)
Comment (axelazaid) [reply models/marts/advertising/fct_ad_performance.sql]: oops, this was supposed to be `... and utm_campaign...` not  `and utm_medium`

File: models/marts/advertising/fct_ad_performance.sql [hunk 3]
```diff
@@ -51,15 +51,15 @@ marketing_channels as (
             when lower(campaign) ilike '%display%' then 'display'
             when lower(campaign) ilike '%shopping%'
                 or lower(campaign) ilike '%cpc%' then 'paid search'
-            when utm_medium is null and utm_medium not in ('fb_retargeted', 'ig_retargeted') 
+            when utm_medium is null and utm_campaign not in ('fb_retargeted', 'ig_retargeted') 
                 then 'direct'
```
Comment (eogilvy12) [inline models/marts/advertising/fct_ad_performance.sql]: @fkalkhalidi should any ads show up as `direct`? on the snowplow side this makes sense but on the ad side we are seeing facebook ads like `utm_campaign = KCP - Store - RT - KCPVC180D (HT_01) (VCEvent) (6-28-18)` show up as direct

```
select distinct
    platform, utm_medium, utm_campaign, campaign, channel
from dbt_adiaz.fct_ad_performance
limit 100
```

File: models/marts/advertising/fct_ad_performance.sql [hunk 4]
```diff
@@ -51,15 +51,15 @@ marketing_channels as (
...
             when lower(utm_campaign) in ('fb_retargeted', 'ig_retargeted') 
                 then 'retargeting'
-            when lower(utm_medium) in ('paid', 'paid_social') 
-                and lower(utm_campaign) not ilike '%amazon%' then 'facebook ads'
             when lower(utm_campaign) ilike '%amazon%' then 'facebook amazon'
+            when lower(utm_medium) in ('paid', 'paid_social')
+                then 'facebook ads'
             when lower(utm_medium) in ('paidsearch', 'cpc', 'shopping') 
                 then 'paid search'
             else null
```
Comment (eogilvy12) [inline models/marts/advertising/fct_ad_performance.sql:64]: @fkalkhalidi a number of ads are showing up with null channel from this categorization - like facebook utm_medium = `prospecting` - is there another way you want these categorized?

**Review by eogilvy12 — COMMENTED** [review]
@axelazaid I just added two questions for faisal to answer before you merge this in. keep an eye on his response and update accordingly :)

File: models/marts/advertising/fct_ad_performance.sql (hunk 3 above)
Comment (fkalkhalidi) [reply models/marts/advertising/fct_ad_performance.sql]: Yes, we shouldn't have any ads show up as direct. 

Also, It seems to me that a number of our social and email campaigns have not been tagged with a utm_medium.

File: models/marts/advertising/fct_ad_performance.sql (hunk 4 above)
Comment (fkalkhalidi) [reply models/marts/advertising/fct_ad_performance.sql:64]: If there are facebook ads that were being categorized as utm_medium = prospecting then we can categorize that with facebook ads.

File: models/marts/advertising/fct_ad_performance.sql (hunk 3 above)
Comment (axelazaid) [reply models/marts/advertising/fct_ad_performance.sql]: @fkalkhalidi there are a number of facebook ads that don't have utm_source or utm_medium, but the majority of utm_campaigns show things like `fb`, `KCP - Store - Cold - LAL - EquipBuyersListLAL 1% (HT_01) (PurchaseEvent) (6-16-18)`,  `prospecting`, `LaborDay`, etc. how should we account for those?

**Pull request merged**